src/parser.js

```diff
@@ -967,12 +969,17 @@ export default class Parser {
                   });
                 }
 
-                if (param.schema && param.schema.format === 'binary') {
-                  return;
+                if (param.schema) {
+                  if (param.schema.format === 'binary') {
+                    return;
+                  }
+
+                  bodyIsPrimitive = (param.schema.type && ['string', 'boolean', 'number'].includes(param.schema.type));
```

Comment:
This should also check for `text/plain` in consumes.

Reply:
Does it have to be `text/plain`? What about `text/html` or `application/my-custom-type`.

Reply:
Yeah. But, let's use the `text/*` check